[fxcop] Settings UI

lmawarid opened the pull request:
## Summary of the Pull Request

Enable FxCop on the Settings.UI project and fixing the associated warnings.

## PR Checklist
* [x] Applies to #5129 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [x] Tests added/passed
![image](https://user-images.githubusercontent.com/16506055/97221069-ad46f880-1789-11eb-80c1-5b9f78185ca2.png)
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

Fixes FxCop warnings including: removing redundant default-value initializations, ensuring Disposable fields are disposed, marking non-instance accessing methods and classes with only static members & methods as static, moving interop methods to NativeMethods class, fixing string-related issues (mainly specifying culture and string normalization), removing unused arguments or suppressing the warning if needed, using safe navigation operator as null argument checks, logging general exceptions that are caught and suppressing ConfigureAwait warnings (which are mostly intended for library code with unpredictable/unspecified execution environments--should be safe to do for UI code).

## Validation Steps Performed

Ensure tests pass and manually validate Settings functionality in Release & Debug builds.

alekhyareddy28 commented on src/core/Microsoft.PowerToys.Settings.UI/Views/ImageResizerPage.xaml.cs:
```diff
@@ -23,30 +26,37 @@ public ImageResizerPage()
-                int rowNum = int.Parse(deleteRowButton.CommandParameter.ToString());
+
+                // Using InvariantCulture since this is internal and expected to be numerical
+                int rowNum = int.Parse(deleteRowButton?.CommandParameter?.ToString(), CultureInfo.InvariantCulture);
```
Could we use `TryParse` instead of throwing and catching general exceptions?

alekhyareddy28 commented on src/core/Microsoft.PowerToys.Settings.UI/Activation/DefaultActivationHandler.cs:
```diff
@@ -18,6 +19,7 @@ public DefaultActivationHandler(Type navElement)
             this.navElement = navElement;
         }
 
+        [SuppressMessage("Reliability", "CA2007:Consider calling ConfigureAwait on the awaited task", Justification = "Warning is intended for library code.")]
```
As Ryan suggested, we can probably use ConfigureAwait(false) unless there is some context in the thread which would be lost otherwise.

lmawarid commented:
@alekhyareddy28 @ryanbodrug-microsoft I addressed the comments yesterday and as far as I can tell there are no issues with the changes so I think this is good for a re-review. 

Regarding the ActivationService classes, I created #7647 to capture that and I can make a separate PR to remove them. I will just see if removing them creates any regressions and maybe wait until tomorrow if Sibille (from Windows Template Studio) can verify if they don't apply to our app (which I'm currently inclined to think).
